File: frontend/src/api/types/song.ts
export type Song = {
  id: number
  title: string | null
  artist: string | null
  src: string
  album: string | null
  file_size: number | null
  user_ids: number[]
  created_at: string
  updated_at: string
}

export type SongCreate = {
  title?: string | null
  artist?: string | null
  album?: string | null
  file: File
}

export type SongUpdate = {
  title?: string | null
  artist?: string | null
  album?: string | null
}

File: backend/app/schemas/song.py
from datetime import datetime
from typing import Optional, List

from pydantic import BaseModel


class SongBase(BaseModel):
    src: str


class SongCreate(BaseModel):
    title: Optional[str] = None
    src: str


class SongUpdate(BaseModel):
    title: Optional[str] = None
    artist: Optional[str] = None
    album: Optional[str] = None


class SongRead(BaseModel):
    id: int
    title: Optional[str] = None
    artist: Optional[str] = None
    src: str
    album: Optional[str] = None
    file_size: Optional[int] = None
    user_ids: List[int] = [] 

    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

